azkaban-common/src/main/java/azkaban/executor/ExecutionLogsDao.java

```diff
@@ -135,12 +135,12 @@ private void uploadLogFile(final DatabaseTransOperator transOperator, final int
     }
   }
 
-  int removeExecutionLogsByTime(final long millis)
+  int removeExecutionLogsByTime(final long millis, final int recordCleanupLimit)
       throws ExecutorManagerException {
     final String DELETE_BY_TIME =
-        "DELETE FROM execution_logs WHERE upload_time < ?";
+        "DELETE FROM execution_logs WHERE upload_time < ? LIMIT ?";
```

Comment:
This is not maintaining retention time for logs as we are deleting only 1000 entries and we might end up with huge number of rows spanning over multiple months over a period of time if the cluster generates more rows as we are restricting ourselves to delete only 24k rows/day. 
Like I pointed out earlier a better would be to probably run this clean up every hour with some batch_size in a loop while there are records to be deleted(i.e older than 12 weeks) and committing each batch.

Reply:
I understand this limit is in config, but deleting fixed number of rows is not deterministic for enforcing retention period